### ISSUE TITLE
Proposal: Add a guide for using CanJS with other libraries

### ISSUE THREAD

**Issue by chasenlehara** [canjs/canjs#4730]
__TLDR:__ add docs for using other libraries inside CanJS and using CanJS components inside other frameworks.

## Motivation

We sometimes get questions about how to use a jQuery plugin or embed a React component inside a CanJS component. It’d be nice to have a guide that answers these questions.

Additionally, it would be nice to show how to use CanJS components inside other frameworks. Having docs on this would help CanJS converts use it in their existing apps, and help show potential converts that it’s easy to start using CanJS in the apps they already maintain.

## Solution

Let’s create a guide that shows how to use other libraries with CanJS. This would show how to embed the following into a CanJS component:

- “Apps”/components from other libraries (Angular, React, Vue, etc.)
- A popular jQuery plugin
- …other popular libraries? [leave a comment please!]

Ideally, we’d walk through one example for each library. This would include instantiating the component, providing data, binding, listening to events, and tearing down the component.

Additionally, we should show the reverse: a CanJS component embedded into popular JS frameworks (Angular, React, etc.). 

It’d be nice to have a working CodePen for each example, but if the library doesn’t make it easy, then a GitHub repo would suffice (à la [webpack-example](https://github.com/canjs/webpack-example)).

## Drawbacks

This adds a maintenance burden with each CanJS release _and_ when other libraries have major releases.

## Alternatives

Instead of writing a guide on canjs.com, write a series of blog or forum posts with each example. This would be a “one-time” expense that we wouldn’t necessarily have to maintain over time.

[Even if we do include this as part of canjs.com, it might not be a bad idea to also publish each one as a blog post for posterity/SEO.]

## Prior art

- [React’s Integrating with Other Libraries](https://reactjs.org/docs/integrating-with-other-libraries.html)
- [Vue.js’s Wrapper Component Example](https://vuejs.org/v2/examples/select2.html)

## Unresolved questions

- Should this go on canjs.com? If so, where?
- This proposes docs for how to integrate other libraries into CanJS and CanJS components into other libraries. Should these be the part of the same doc, or two separate guides?
